numba/np/arrayobj.py

```diff
@@ -6376,3 +6376,53 @@ def take_along_axis_impl(arr, indices, axis):
             return _take_along_axis_impl(arr, indices, axis, Ni, Nk,
                                          indices_broadcast_shape)
     return take_along_axis_impl
+
+
+@overload(np.nan_to_num)
+def nan_to_num_impl(x, copy=True, nan=0.0):
+    if isinstance(x, types.Number):
+        if isinstance(x, types.Integer):
+            # Integers do not have nans or infs
+            def impl(x, copy=True, nan=0.0):
+                return x
+
+        elif isinstance(x, types.Float):
+            def impl(x, copy=True, nan=0.0):
+                if np.isnan(x):
+                    return nan
+                elif np.isneginf(x):
+                    return np.finfo(np.float64).min
+                elif np.isposinf(x):
+                    return np.finfo(np.float64).max
+                return x
+        else:
+            raise errors.TypingError("Only Integers and Floats are accepted")
```

Comment:
Any reason for not supporting complex values as well? I'm fine with restricting the PR to only int and float values, but you could support complex values as well by reusing the code from float:

```python
if isinstance(x, types.Integer):
    ...
elif isinstance(x, types.Float):
    ...
elif isinstance(x, types.Complex):
    def impl(x, copy=True, nan=0.0):
        r = np.nan_to_num(x.real)
        i = np.nan_to_num(x.imag)
        return complex(r, i)
```